packages/babel-plugin-transform-modules-commonjs/src/index.ts

```diff
@@ -81,14 +81,35 @@ export default declare((api, options) => {
         (path.parentPath.isObjectProperty({ value: path.node }) &&
           path.parentPath.parentPath.isObjectPattern()) ||
         path.parentPath.isAssignmentExpression({ left: path.node }) ||
-        path.isAssignmentExpression({ left: path.node })
+        path.isAssignmentExpression({ left: path.node }) ||
+        path.parentPath.isUpdateExpression()
       ) {
         return;
       }
 
       path.replaceWith(getAssertion(localName));
     },
 
+    UpdateExpression(path) {
```

Comment:
Q: What is this change for?

Reply:
This seems to be a redundant change.
All tests passed without it.

Reply:
I mean the line 85

Reply:
line 93-111 to prevent the following case:

```javascript
module++
```

```javascript
function () {
    throw new Error("The CommonJS '" + "module" + "' variable is not available in ES6 modules." + "Consider setting setting sourceType:script or sourceType:unambiguous in your " + "Babel config for this file.");
}()++;
```

Reply:
instead, we expect the following result
```javascript
module += function () {
    throw new Error("The CommonJS '" + "module" + "' variable is not available in ES6 modules." + "Consider setting setting sourceType:script or sourceType:unambiguous in your " + "Babel config for this file.");
}();
```

Reply:
Oh well, both those pieces of code have the same behavior, so whatever we get with less code is ok.